services/web/server/src/simcore_service_webserver/login/storage.py

```diff
@@ -90,12 +93,12 @@ async def create_confirmation(
             assert code == c  # nosec
             return confirmation
 
-    async def get_confirmation(self, filter_dict) -> Optional[ConfirmationTokenDict]:
+    async def get_confirmation(self, filter_dict) -> ConfirmationTokenDict | None:
         if "user" in filter_dict:
             filter_dict["user_id"] = filter_dict.pop("user")["id"]
         async with self.pool.acquire() as conn:
             confirmation = await _sql.find_one(conn, self.confirm_tbl, filter_dict)
-            return ConfirmationTokenDict(**confirmation) if confirmation else None
+            return ConfirmationTokenDict(**confirmation) if confirmation else None  # type: ignore
```

Comment:
This looks fishy. What about reducing indentation by 4 spaces? Would it then not always return instead of returning `None` and or `ConfirmationTokenDict`

Reply:
Currently, I do not know how to quickly fix this without spending more time understanding the code. I will remove the ignore for now and come to it later.